src/components/InteractionLayer/index.js

```diff
@@ -68,6 +68,32 @@ class InteractionLayer extends React.Component {
     this.syncZoomingState();
   }
 
+  componentWillReceiveProps(nextProps) {
+    const { subDomain: prevSuDomain } = this.props;
```

Comment:
Would be nice to check if ruler is active first, then we wouldn't do this computation and render every time. If you merge it at least mark it somewhere that it could be optimised (issue or something).

Reply:
you are absolutely right, updated